jpype/protocol.py

```diff
@@ -17,35 +17,29 @@
 _JArray = _jpype._JArray
 _JBuffer = _jpype._JBuffer
 
-
 if sys.version_info < (3, 8):
-    # Before 3.8 we have to use attribute type converters
+    from typing_extensions import Protocol, runtime_checkable
     from collections.abc import Sequence, Mapping
 
-    class _AttributeMeta(type):
-        def __instancecheck__(self, obj):
-            return hasattr(obj, self._attrib)
-    SupportsPath = _AttributeMeta("SupportsPath", (object,), {
-                                  "_attrib": "__fspath__"})
-    SupportsIndex = _AttributeMeta("SupportsIndex", (object,), {
-                                   "_attrib": "__index__"})
-    SupportsFloat = _AttributeMeta("SupportsFlaot", (object,), {
-                                   "_attrib": "__float__"})
-else:
-    from typing import Protocol, runtime_checkable, abstractmethod
-    from typing import SupportsIndex, SupportsFloat
-    from typing import Sequence, Mapping
+    @runtime_checkable
+    class SupportsFloat(Protocol):
```

Comment:
I think this is not needed, Python >= 3.5 should have it in the standard library `typing` module